### PR TITLE
Example Values - Distances - PR created as a place to talk about code structure

### PR DESCRIPTION
**Pull request by adeptofvoltron**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "pcf/valueobject",
   "description": "todo",
-  "type": "metapackage",
+  "type": "metcomapackage",
```
Comment (pradzikowski) [inline composer.json]: Is it intentional ?

File: phpunit.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pradzikowski) [inline phpunit.xml]: You forgot merge code from PCF to your local fork ?

**Review by pradzikowski — CHANGES_REQUESTED** [review]

File: src/Distance/AbstractDistance.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if (false === is_numeric($quality)) {
+            throw new \InvalidArgumentException('quality expect to be a number');
+        }
+
+        $this->quality = (float) $quality;
+    }
+
+    public function isEqualTo(ValueObjectInterface $compare): bool
+    {
+        $className = get_called_class();
```
Comment (pradzikowski) [inline src/Distance/AbstractDistance.php]: 👍

File: src/Distance/Comparator.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+     * @param AbstractDistance $right
+     *
+     * @return bool
+     */
+    public static function compare(AbstractDistance $left, AbstractDistance $right): bool
+    {
+        $tmpDistance = Converter::convert($right, get_class($left));
+        return $left->isEqualTo($tmpDistance);
+    }
+}
```
Comment (pradzikowski) [inline src/Distance/Comparator.php]: Missing line after bracket.

File: src/Distance/Comparator.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    /**
+     * @param AbstractDistance $left
+     * @param AbstractDistance $right
+     *
+     * @return bool
+     */
+    public static function compare(AbstractDistance $left, AbstractDistance $right): bool
+    {
+        $tmpDistance = Converter::convert($right, get_class($left));
+        return $left->isEqualTo($tmpDistance);
```
Comment (pradzikowski) [inline src/Distance/Comparator.php]: missing line before return

File: src/Distance/Converter.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * @param $feets
+     * @return float
+     */
+    private static function feetToMeter($feets): float
+    {
+        return $feets * 0.3048;
+    }
+}
```
Comment (pradzikowski) [inline src/Distance/Converter.php]: missing line after bracket :P

**Comment by adeptofvoltron** [comment]
@pradzikowski first start about files structure. Is it ok for you. Or You would rather wish to push all comparators/converters to own directory

File: composer.json (hunk 1 above)
Comment (adeptofvoltron) [reply composer.json]: done

File: phpunit.xml (hunk 2 above)
Comment (adeptofvoltron) [reply phpunit.xml]: done - nope just commited it before we merged new .gitignore rule

File: src/Distance/AbstractDistance.php (hunk 3 above)
Comment (adeptofvoltron) [reply src/Distance/AbstractDistance.php]: ok I will use static::class

File: src/Distance/Comparator.php (hunk 4 above)
Comment (adeptofvoltron) [reply src/Distance/Comparator.php]: done

File: src/Distance/Comparator.php (hunk 5 above)
Comment (adeptofvoltron) [reply src/Distance/Comparator.php]: done

File: src/Distance/Converter.php (hunk 6 above)
Comment (adeptofvoltron) [reply src/Distance/Converter.php]: done

**Comment by pradzikowski** [comment]
@adeptofvoltron there is a conflict.

**Review by pradzikowski — APPROVED** [review]

**Comment by adeptofvoltron** [comment]
@pradzikowski Done

to  sum-up our speech during PCF meeting. In these repo we will not keep any converters/comparators etc... it is only for VO. All these utils will have their own repo in the future.

**Pull request merged**